src/device.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013-2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use gdk_sys;
+use glib::object::IsA;
+use glib::translate::*;
+use Device;
+use Surface;
+use TimeCoord;
+
+use std::mem;
+use std::ptr;
+
+impl Device {
+    pub fn get_history<P: IsA<Surface>>(
+        &self,
+        surface: &P,
+        start: u32,
+        stop: u32,
+    ) -> Option<Vec<TimeCoord>> {
+        unsafe {
+            let mut events = ptr::null_mut();
+            let mut n_events = mem::MaybeUninit::uninit();
+            let ret: bool = from_glib(gdk_sys::gdk_device_get_history(
+                self.to_glib_none().0,
+                surface.as_ref().to_glib_none().0,
+                start,
+                stop,
+                &mut events,
+                n_events.as_mut_ptr(),
+            ));
+            if ret {
```

Comment:
Shouldn't it be:
```suggestion
            if !ret {
```

Reply:
Yes, see also https://github.com/gtk-rs/gdk/pull/314#discussion_r323085555 :)

Reply:
I wonder how I could fail this one too... (bad c/p I guess...)